test/client_mqtt5.js

```diff
@@ -0,0 +1,538 @@
+'use strict'
+
+var mqtt = require('..')
+var abstractClientTests = require('./abstract_client')
+var Buffer = require('safe-buffer').Buffer
+var MqttServer = require('./server').MqttServer
+var assert = require('chai').assert
+var serverBuilder = require('./server_helpers_for_client_tests').serverBuilder
+var ports = require('./helpers/port_list')
+
+describe('MQTT 5.0', function () {
+  var server = serverBuilder().listen(ports.PORTAND115)
+  var config = { protocol: 'mqtt', port: ports.PORTAND115, protocolVersion: 5, properties: { maximumPacketSize: 200 } }
+
+  abstractClientTests(server, config)
+
+  // var server = serverBuilder().listen(ports.PORTAND115)
+
+  var topicAliasTests = [
+    {properties: {}, name: 'should allow any topicAlias when no topicAliasMaximum provided in settings'},
+    {properties: { topicAliasMaximum: 15 }, name: 'should not allow topicAlias > topicAliasMaximum when topicAliasMaximum provided in settings'}
+  ]
+
+  topicAliasTests.forEach(function (test) {
+    it(test.name, function (done) {
+      this.timeout(15000)
+      server.once('client', function (serverClient) {
+        serverClient.on('publish', function (packet) {
+          if (packet.properties && packet.properties.topicAlias) {
+            done(new Error('Packet should not have topicAlias'))
+            return false
+          } else {
+            serverClient.end(done)
+          }
+        })
+      })
+      var opts = {host: 'localhost', port: ports.PORTAND115, protocolVersion: 5, properties: test.properties}
+      var client = mqtt.connect(opts)
+      client.publish('t/h', 'Message', { properties: { topicAlias: 22 } })
+    })
+  })
+
+  it('should throw an error if there is Auth Data with no Auth Method', function (done) {
+    this.timeout(5000)
+    var client
+    var opts = {host: 'localhost', port: ports.PORTAND115, protocolVersion: 5, properties: { authenticationData: Buffer.from([1, 2, 3, 4]) }}
+    console.log('client connecting')
+    client = mqtt.connect(opts)
+    client.on('error', function (error) {
+      console.log('error hit')
+      assert.strictEqual(error.message, 'Packet has no Authentication Method')
+      // client will not be connected, so we will call done.
+      assert.isTrue(client.disconnected, 'validate client is disconnected')
+      client.end(true)
+      done()
+    })
+  })
+
+  it('auth packet', function (done) {
+    this.timeout(15000)
```

Comment:
>this.timeout(15000) [](start = 4, length = 19)

These are all local server connections?  Does it really need this long of a timeout?

Reply:
they are local server connections but still need this timeout.

---
In reply to: [407791930](https://github.com/mqttjs/MQTT.js/pull/1076#discussion_r407791930) [](ancestors = 407791930)